test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.cs

```diff
@@ -99,38 +148,47 @@ public async Task CheckEnrichmentWhenSampling()
         await CheckEnrichment(new AlwaysOnSampler(), true, this.url).ConfigureAwait(false);
     }
 
-    private async Task HttpOutCallsAreCollectedSuccessfullyBodyAsync(HttpTestData.HttpOutTestCase tc, bool enableTracing, bool enableMetrics)
+    private static async Task HttpOutCallsAreCollectedSuccessfullyBodyAsync(
+        string host,
+        int port,
+        HttpTestData.HttpOutTestCase tc,
+        bool enableTracing,
+        bool enableMetrics,
+        HttpSemanticConvention? semanticConvention = null)
     {
         bool enrichWithHttpWebRequestCalled = false;
         bool enrichWithHttpWebResponseCalled = false;
         bool enrichWithHttpRequestMessageCalled = false;
         bool enrichWithHttpResponseMessageCalled = false;
         bool enrichWithExceptionCalled = false;
 
-        using var serverLifeTime = TestHttpServer.RunServer(
-            (ctx) =>
-            {
-                ctx.Response.StatusCode = tc.ResponseCode == 0 ? 200 : tc.ResponseCode;
-                ctx.Response.OutputStream.Close();
-            },
-            out var host,
-            out var port);
-
-        tc.Url = HttpTestData.NormalizeValues(tc.Url, host, port);
```

Comment:
@matt-hensley This line here was probably causing the most pain. The test cases are static. So the first run through would modify the Url and then subsequent tests would send to the server port from the first run. Fixing this fixed most of the test failures 🤣